frontend/lib/identification/card_detail_view/verification_code_view.dart

```diff
@@ -57,32 +65,69 @@ class VerificationCodeViewState extends State<VerificationCodeView> {
             );
             qrCode.make();
 
-            return ConstrainedBox(
-              constraints: const BoxConstraints(maxWidth: 600, maxHeight: 600),
-              child: Material(
-                clipBehavior: Clip.hardEdge,
-                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
-                child: Stack(
-                  children: [
-                    Padding(
-                      padding: EdgeInsets.all(padding),
-                      child: qr.QrImage.withQr(
-                        qr: qrCode,
-                        version: qr.QrVersions.auto,
-                        foregroundColor: Theme.of(context).textTheme.bodyMedium?.color,
-                        gapless: false,
+            return isVerificationExpired
+                ? IconButton(
+                    icon: const Icon(Icons.refresh),
+                    color: Theme.of(context).appBarTheme.backgroundColor,
+                    onPressed: () {
+                      verifyCard(otpCode, userCode, context);
+                    },
+                  )
```

Comment:
I haven't built the app and looked at it yet, but I suppose, we need some textual explanation of what's going on here.

Reply:
i propose a text above with last commit